src/Memory.cs

```diff
@@ -62,6 +62,11 @@ public Memory(IStore store, uint minimum = 0, uint maximum = uint.MaxValue)
         /// </summary>
         public uint Maximum { get; private set; }
 
+        /// <summary>
+        /// The memory encoding for string operations. 
+        /// </summary>
+        public Encoding MemoryEncoding { get; set; } = Encoding.UTF8;
```

Comment:
Would perhaps a `Encoding` parameter to `ReadString` and `WriteString` defaulted to `Encoding.UTF8` would be a little more ergonomic for anyone wishing to read different string encodings from the same linear memory (although not as likely as scenario)?

Reply:
Related, we should update the doc comments for `ReadString` and `WriteString` so as to not be UTF-8 specific.